src/cmd/sanity_test/sanity_tests.go

```diff
@@ -85,29 +92,20 @@ func checkEmailRestoration(
 
 		values := result.GetValue()
 
-		// recursive restore folder discovery before proceeding with tests
 		for _, v := range values {
-			var (
-				itemID              = ptr.Val(v.GetId())
-				itemName            = ptr.Val(v.GetDisplayName())
-				ictx                = clues.Add(ctx, "item_id", itemID, "item_name", itemName)
-				folderTime, hasTime = mustGetTimeFromName(ctx, itemName)
-			)
-
-			if !isWithinTimeBound(ictx, startTime, folderTime, hasTime) {
-				continue
-			}
+			itemName := ptr.Val(v.GetDisplayName())
 
-			// if we found the folder to testt against, back out of this loop.
 			if itemName == folderName {
 				restoreFolder = v
 				continue
 			}
 
-			// otherwise, recursively aggregate all child folders.
-			getAllSubFolder(ctx, client, testUser, v, itemName, itemCount)
+			if itemName == dataFolder || itemName == baseBackupFolder {
+				// otherwise, recursively aggregate all child folders.
+				getAllSubFolder(ctx, client, testUser, v, itemName, dataFolder, itemCount)
 
-			itemCount[itemName] = ptr.Val(v.GetTotalItemCount())
+				itemCount[itemName], _ = ptr.ValOK(v.GetTotalItemCount())
```

Comment:
If you're swallowing the boolean on `ptr.ValOK` you should use `ptr.Val` instead.
```suggestion
				itemCount[itemName] = ptr.Val(v.GetTotalItemCount())
```